api/src/com/cloud/naming/ResourceNamingPolicyManager.java

```diff
@@ -0,0 +1,32 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.naming;
+
+import com.cloud.utils.component.Manager;
+
+public interface ResourceNamingPolicyManager extends Manager {
+
+
+    /**
+     * Finds a resource naming policy among the registered ones by its class
+     * @param policyClass
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    public <T extends ResourceNamingPolicy> T getPolicy(Class<T> policyClass);
```

Comment:
Let’s check if I understood this method.
It will look up for a naming policy that is registered given a naming policy class, right?

The registered policy is an instance of that class that is sent as a parameter, right? If I already know the class I need, why look for an instance of it?

What about creating singletons those classes using spring? And then, just injecting them where needed. Sorry if overlooked something, but have you tried to design some solution that uses IoC instead of creating our own?